storage/migrate.go

```diff
@@ -0,0 +1,155 @@
+package storage
+
+import (
+	"context"
+	"strconv"
+
+	_ "github.com/filecoin-project/sentinel-visor/storage/migrations"
+	"github.com/go-pg/migrations/v8"
+	"github.com/go-pg/pg/v10"
+	"golang.org/x/xerrors"
+)
+
+// GetSchemaVersions returns the schema version in the database and the latest schema version defined by the available
+// migrations.
+func (d *Database) GetSchemaVersions(ctx context.Context) (int64, int64, error) {
+	// If we're already connected then use that connection
+	if d.DB != nil {
+		return getSchemaVersions(ctx, d.DB)
+	}
+
+	// Temporarily connect
+	db, err := connect(ctx, d.opt)
+	if err != nil {
+		return 0, 0, xerrors.Errorf("connect: %w", err)
+	}
+	defer db.Close()
+	return getSchemaVersions(ctx, db)
+}
+
+// getSchemaVersions returns the schema version in the database and the schema version defined by the available
+// migrations.
+func getSchemaVersions(ctx context.Context, db *pg.DB) (int64, int64, error) {
+	// Run the migration init to ensure we always have a migrations table
+	_, _, err := migrations.Run(db, "init")
+	if err != nil {
+		return 0, 0, xerrors.Errorf("migration table init: %w", err)
+	}
+
+	dbVersion, err := migrations.Version(db)
+	if err != nil {
+		return 0, 0, xerrors.Errorf("unable to determine schema version: %w", err)
+	}
+
+	// Current desired schema version is based on the highest migration version
+	var desiredVersion int64
+	ms := migrations.DefaultCollection.Migrations()
+	for _, m := range ms {
+		if m.Version > desiredVersion {
+			desiredVersion = m.Version
+		}
+	}
+
+	return dbVersion, desiredVersion, nil
+}
+
+// MigrateSchema migrates the database schema to the latest version based on the list of migrations available
+func (d *Database) MigrateSchema(ctx context.Context) error {
```

Comment:
commentary: would be great if the below logic could be reused by `MigrateSchemaTo`

Reply:
It's not easy to because they have different logic between getting the schema versions and taking the lock